rust/tw_bitcoin/src/modules/legacy/scripts.rs

```diff
@@ -0,0 +1,249 @@
+use crate::aliases::*;
+use bitcoin::PublicKey;
+use std::ffi::{c_char, CStr};
+use tw_memory::ffi::c_byte_array::CByteArray;
+use tw_memory::ffi::c_byte_array_ref::CByteArrayRef;
+use tw_misc::try_or_else;
+use tw_proto::Bitcoin::Proto as LegacyProto;
+use tw_proto::BitcoinV2::Proto;
+
+#[no_mangle]
+#[deprecated]
+// Builds the P2PKH scriptPubkey.
+pub unsafe extern "C" fn tw_build_p2pkh_script(
+    _satoshis: i64,
+    pubkey: *const u8,
+    pubkey_len: usize,
+) -> CByteArray {
+    // Convert Recipient
+    let slice = try_or_else!(
+        CByteArrayRef::new(pubkey, pubkey_len).as_slice(),
+        CByteArray::null
+    );
+    let recipient = try_or_else!(PublicKey::from_slice(slice), CByteArray::null);
+
+    //let tx_out = TxOutputP2PKH::new(satoshis as u64, recipient);
+    let output = Proto::Output {
+        value: _satoshis as u64,
+        to_recipient: ProtoOutputRecipient::builder(Proto::mod_Output::OutputBuilder {
+            variant: ProtoOutputBuilder::p2pkh(Proto::ToPublicKeyOrHash {
+                to_address: Proto::mod_ToPublicKeyOrHash::OneOfto_address::pubkey(
+                    recipient.to_bytes().into(),
+                ),
+            }),
+        }),
+    };
+
+    let res = try_or_else!(
+        crate::modules::transactions::OutputBuilder::utxo_from_proto(&output),
+        CByteArray::null
+    );
+
+    // Prepare and serialize protobuf structure.
+    let proto = LegacyProto::TransactionOutput {
+        value: res.value as i64,
+        script: res.script_pubkey,
+        spendingScript: Default::default(),
+    };
+
+    let serialized = tw_proto::serialize(&proto).expect("failed to serialized transaction output");
```

Comment:
This should never fail until we use `Writer<Vec<u8>>` protobuf writer.
Probably, we can make `tw_proto::serialize` never fail by using `expect` internally 🤔

Reply:
At this to a Todo list?